scrapy/core/scheduler.py

```diff
@@ -256,6 +333,10 @@ def next_request(self) -> Optional[Request]:
         Increment the appropriate stats, such as: ``scheduler/dequeued``,
         ``scheduler/dequeued/disk``, ``scheduler/dequeued/memory``.
         """
+        delayed_request = self.dpqs.pop()
+        if delayed_request:
+            self.stats.inc_value('scheduler/dequeued/delayed/memory', spider=self.spider)
+            self._enqueue_request(delayed_request)
```

Comment:
I am also thinking this should be a loop, pop and enqueue all delayed requests whose time has passed.